### PR TITLE
Show full version on package details page

### PR DESCRIPTION
**Pull request by xavierdecoster**
Only display the full version in the title.
Use NormalizedVersion anywhere else.

image

### CODE REVIEW

**Comment by joelverhagen** [comment]
Why no full version in the version list at the bottom of the page? 

Also, is this full version visible on web UI package upload or package edit?

**Comment by xavierdecoster** [comment]
> Why no full version in the version list at the bottom of the page?

Full Version may be quite large (I've seen people using metadata parts of 50+ characters), so wouldn't want to clutter the table. As the URL for those hyperlink uses NormalizedVersion anyway, I figured it made sense to keep those as is.

Same for the version in the left side bar, don't want it to flow through in other text.

> Also, is this full version visible on web UI package upload or package edit?

That is already the case. Try uploading one ;)

**Comment by joelverhagen** [comment]
Sounds good. Thanks.

File: src/NuGetGallery/ViewModels/PackageViewModel.cs [hunk 1]
```diff
@@ -3,20 +3,34 @@
...
             _package = package;
+
+            NuGetVersion nugetVersion;
+            if(NuGetVersion.TryParse(package.Version, out nugetVersion))
+            {
+                _fullVersion = nugetVersion.ToFullString();
+            }
+            else
+            {
+                _fullVersion = string.Empty;
```
Comment (scottbommarito) [inline src/NuGetGallery/ViewModels/PackageViewModel.cs]: If this parsing fails, the UI will show no version, which shouldn't be possible.

A couple options:
1 - fall back to what `Version` shows
2 - throw an exception and fail
3 - print `package.Version` directly ([which we currently do](https://github.com/NuGet/NuGetGallery/blob/29b3aa1e57c2f4aa3e990b198d7fc9dd7029d724/src/NuGetGallery.Core/NuGetVersionExtensions.cs#L17))

File: src/NuGetGallery/ViewModels/PackageViewModel.cs [hunk 2]
```diff
@@ -3,20 +3,34 @@
...
         private readonly Package _package;
         private string _pendingTitle;
+        private string _fullVersion;
 
         public PackageViewModel(Package package)
         {
             _package = package;
+
+            NuGetVersion nugetVersion;
+            if(NuGetVersion.TryParse(package.Version, out nugetVersion))
```
Comment (scottbommarito) [inline src/NuGetGallery/ViewModels/PackageViewModel.cs]: Should we move this to [NuGetVersionExtensions](https://github.com/NuGet/NuGetGallery/blob/29b3aa1e57c2f4aa3e990b198d7fc9dd7029d724/src/NuGetGallery.Core/NuGetVersionExtensions.cs)?

`NuGetFullVersioner`?

File: src/NuGetGallery/ViewModels/PackageViewModel.cs (hunk 1 above)
Comment (xavierdecoster) [inline src/NuGetGallery/ViewModels/PackageViewModel.cs]: Addressed with https://github.com/NuGet/NuGetGallery/pull/3887/commits/2767721f09aa40b9063845756dd6250c4f94a147

File: src/NuGetGallery/ViewModels/PackageViewModel.cs (hunk 2 above)
Comment (xavierdecoster) [inline src/NuGetGallery/ViewModels/PackageViewModel.cs]: Addressed with https://github.com/NuGet/NuGetGallery/pull/3887/commits/2767721f09aa40b9063845756dd6250c4f94a147

**Pull request merged**